app/views/container_topology/show.html.haml

```diff
@@ -1,84 +1,109 @@
+        %svg.kube-topology
+          %g.EntityLegend.Containers
+            %circle{:r => "17"}
+            -# pficon-replicator
+            %text{:y => "8"} &#xE624;
+        %label Replicators
+      %kubernetes-topology-icon{:kind => "ContainerGroup", "tooltip-placement" => "bottom", "tooltip" => "{{legendTooltip}}"}
+        %svg.kube-topology
+          %g.EntityLegend.Containers.Pod
+            %circle{:r => "17"}
+            -# fa-cubes
+            %text{:x => "1", :y => "6"} &#xF1B3;
+        %label Pods
+      %kubernetes-topology-icon{:kind => "Container", "tooltip-placement" => "bottom", "tooltip" => "{{legendTooltip}}"}
+        %svg.kube-topology
+          %g.EntityLegend.Containers.Container
+            %circle{:r => "17"}
+            -# fa-cube
+            %text{:y => "7"} &#xF1B2;
+        %label Containers
+      %kubernetes-topology-icon{:kind => "ContainerService", "tooltip-placement" => "bottom", "tooltip" => "{{legendTooltip}}"}
+        %svg.kube-topology
+          %g.EntityLegend.Containers
+            %circle{:r => "17"}
+            -# pficon-service
+            %text{:y => "8"} &#xE61E;
+        %label Services
+      %kubernetes-topology-icon{:kind => "ContainerRoute", "tooltip-placement" => "bottom", "tooltip" => "{{legendTooltip}}"}
+        %svg.kube-topology
+          %g.EntityLegend.Containers
+            %circle{:r => "17"}
+            -# pficon-route
+            %text{:y => "8"} &#xE625;
+        %label Routes
+      %kubernetes-topology-icon{:kind => "ContainerNode", "tooltip-placement" => "bottom", "tooltip" => "{{legendTooltip}}"}
+        %svg.kube-topology
+          %g.EntityLegend.Containers
+            %circle{:r => "17"}
+            -# pficon-container-node
+            %text{:y => "9"} &#xE621;
+        %label Nodes
+      %kubernetes-topology-icon{:kind => "Vm", "tooltip-placement" => "bottom", "tooltip" => "{{legendTooltip}}"}
+        %svg.kube-topology
+          %g.EntityLegend.Infra
+            %circle{:r => "17"}
+            -# pficon-screen
+            %text{:y => "9"} &#xE600;
+        %label VMs
+      %kubernetes-topology-icon{:kind => "Host", "tooltip-placement" => "bottom", "tooltip" => "{{legendTooltip}}"}
+        %svg.kube-topology
+          %g.EntityLegend.Infra
+            %circle{:r => "17"}
+            -# pficon-cluster
+            %text{:y => "9"} &#xE620;
+        %label Hosts
 
   .alert.alert-info.alert-dismissable
     %button.close{"aria-hidden" => "true", "data-dismiss" => "alert", :type => "button"}
       %span.pficon.pficon-close
     %span.pficon.pficon-info
-    %strong  Click on the legend to show/hide entities, and double click on the entities in the graph to navigate to their summary pages. 
+    %strong  Click on the legend to show/hide entities, and double click on the entities in the graph to navigate to their summary pages.
   %kubernetes-topology-graph{:items => "items", :relations => "relations", :kinds => "kinds"}
+:javascript
```

Comment:
@abonas The JS has been removed and we are now calling the new function miqHideSearchClearButton